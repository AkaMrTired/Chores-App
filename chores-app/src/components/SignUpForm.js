/* eslint-disable jsx-a11y/label-has-associated-control */
import axios from "axios";
import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { useUserAuth } from "../context/UserAuthContext";

const SignUpForm = () => {
  const initialState = {
    fields: {
      familyName: "",
      yourName: "",
      yourEmail: "",
      password: "",
      confirmPassword: "",
    },
  };
  const [fields, setFields] = useState(initialState.fields);
  const [error, setError] = useState();

  const { signUp } = useUserAuth();
  const navigate = useNavigate();

  const registration = async (event) => {
    event.preventDefault();
    if (fields.password === fields.confirmPassword) {
      try {
        await signUp(fields.yourEmail, fields.password);
        await axios
          .post(`http://localhost:3300/family`, {
            familyName: fields.familyName,
          })
          .then((response) => {
            const { familyID } = response.data;
            localStorage.setItem("familyID", JSON.stringify(familyID));
          })
          .catch((e) => {
            console.log(e);
            event.preventDefault();
          });
        navigate("/parentdashboard");
      } catch (e) {
        setError(e.message);
      }
    } else {
      setError("Those passwords did not match, please try again");
    }
  };

  const handleFieldChange = (event) => {
    event.preventDefault();
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="sign-up-container">
      <h1>Sign Up</h1>
      <div>
        <form onSubmit={registration}>
          <label htmlFor="familyName">Family Name </label>
          <input
            name="familyName"
            required
            type="text"
            placeholder="e.g. The Cool Gang"
            value={fields.familyName}
            onChange={handleFieldChange}
          />

          <label htmlFor="yourName">Your Name </label>
          <input
            name="yourName"
            required
            type="text"
            placeholder="e.g Mommy"
            value={fields.yourName}
            onChange={handleFieldChange}
          />

          <label htmlFor="yourEmail">Your Email</label>
          <input
            type="email"
            name="yourEmail"
            required
            placeholder="e.g example@example.com"
            value={fields.email}
            onChange={handleFieldChange}
          />

          <label htmlFor="password">
            Password (minimum 6 characters, must contain a letter and a number)
          </label>
          <input
            type="password"
            name="password"
            data-testid="password"
            required
            placeholder="*******"
            minLength="6"
            pattern="^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{5,}$"
            value={fields.password}
            onChange={handleFieldChange}
          />

          <label htmlFor="confirmPassword">Confirm Password </label>
          <input
            type="password"
            name="confirmPassword"
            required
            placeholder="*******"
            value={fields.confirmPassword}
            onChange={handleFieldChange}
          />
          {!!error && <p>{error}</p>}

          <button type="submit">Create Account</button>
        </form>
      </div>
      <p>
        You will be able to invite more family members once you make your
        account.
      </p>
    </div>
  );
};

export default SignUpForm;
