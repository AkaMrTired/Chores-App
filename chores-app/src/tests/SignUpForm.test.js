import { render, screen } from "@testing-library/react";
import React from "react";
import SignUpForm from "../components/SignUpForm";

describe("the sign up form", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<SignUpForm />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<SignUpForm />);
    expect(screen.getByText(/Sign Up/i)).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /Create Account/i })
    ).toBeInTheDocument();
    expect(screen.getByText(/family name/i)).toBeInTheDocument();
    expect(screen.getByText(/your name/i)).toBeInTheDocument();
    expect(screen.getByText(/email/i)).toBeInTheDocument();
    expect(screen.getByTestId("password")).toBeInTheDocument();
    expect(screen.getByText(/confirm password/i)).toBeInTheDocument();
    expect(
      screen.getByText(
        /You will be able to invite more family members once you make your account./i
      )
    ).toBeInTheDocument();
  });
});
