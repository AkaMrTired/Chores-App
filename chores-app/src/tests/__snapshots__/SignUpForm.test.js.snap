// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the sign up form matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sign-up-container"
  >
    <h1>
      Sign Up
    </h1>
    <div>
      <form>
        <label
          for="familyName"
        >
          Family Name 
        </label>
        <input
          name="familyName"
          placeholder="e.g. The Cool Gang"
          required=""
          type="text"
          value=""
        />
        <label
          for="yourName"
        >
          Your Name 
        </label>
        <input
          name="yourName"
          placeholder="e.g Mommy"
          required=""
          type="text"
          value=""
        />
        <label
          for="yourEmail"
        >
          Your Email
        </label>
        <input
          name="yourEmail"
          placeholder="e.g example@example.com"
          required=""
          type="email"
          value=""
        />
        <label
          for="password"
        >
          Password (minimum 5 characters, must contain a letter and a number)
        </label>
        <input
          data-testid="password"
          minlength="5"
          name="password"
          pattern="^(?=.*[A-Za-z])(?=.*\\\\d)[A-Za-z\\\\d]{5,}$"
          placeholder="*******"
          required=""
          type="password"
          value=""
        />
        <label
          for="confirmPassword"
        >
          Confirm Password 
        </label>
        <input
          name="confirmPassword"
          placeholder="*******"
          required=""
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Create Account
        </button>
      </form>
    </div>
    <p>
      You will be able to invite more family members once you make your account.
    </p>
  </div>
</DocumentFragment>
`;
